pkg/plugin/logrus_logger.go

```diff
@@ -0,0 +1,130 @@
+// are expected to be strings, and values can be any type.
+func argsToFields(args ...interface{}) logrus.Fields {
+	var (
+		fields = make(map[string]interface{})
+		field  = ""
+	)
+
+	for _, arg := range args {
+		if field == "" {
+			field = arg.(string)
+		} else {
+			fields[field] = arg
+			field = ""
+		}
+	}
+
+	return logrus.Fields(fields)
+}
+
+// Trace emits a message and key/value pairs at the TRACE level
+func (l *logrusLogger) Trace(msg string, args ...interface{}) {
+	l.Debug(msg, args...)
+}
+
+// Debug emits a message and key/value pairs at the DEBUG level
+func (l *logrusLogger) Debug(msg string, args ...interface{}) {
+	l.impl.WithFields(argsToFields(args...)).Debug(msg)
+}
+
+// Info emits a message and key/value pairs at the INFO level
+func (l *logrusLogger) Info(msg string, args ...interface{}) {
+	l.impl.WithFields(argsToFields(args...)).Info(msg)
+}
+
+// Warn emits a message and key/value pairs at the WARN level
+func (l *logrusLogger) Warn(msg string, args ...interface{}) {
+	l.impl.WithFields(argsToFields(args...)).Warn(msg)
+}
+
+// Error emits a message and key/value pairs at the ERROR level
+func (l *logrusLogger) Error(msg string, args ...interface{}) {
+	l.impl.WithFields(argsToFields(args...)).Error(msg)
+}
+
+// IsTrace indicates if TRACE logs would be emitted. This and the other Is* guards
+// are used to elide expensive logging code based on the current level.
+func (l *logrusLogger) IsTrace() bool {
+	return l.IsDebug()
+}
+
+// IsDebug indicates if DEBUG logs would be emitted. This and the other Is* guards
+// are used to elide expensive logging code based on the current level.
+func (l *logrusLogger) IsDebug() bool {
+	return l.level <= logrus.DebugLevel
+}
+
+// IsInfo indicates if INFO logs would be emitted. This and the other Is* guards
+// are used to elide expensive logging code based on the current level.
+func (l *logrusLogger) IsInfo() bool {
+	return l.level <= logrus.InfoLevel
+}
+
+// IsWarn indicates if WARN logs would be emitted. This and the other Is* guards
+// are used to elide expensive logging code based on the current level.
+func (l *logrusLogger) IsWarn() bool {
+	return l.level <= logrus.WarnLevel
+}
+
+// IsError indicates if ERROR logs would be emitted. This and the other Is* guards
+// are used to elide expensive logging code based on the current level.
+func (l *logrusLogger) IsError() bool {
+	return l.level <= logrus.ErrorLevel
+}
+
+// With creates a sublogger that will always have the given key/value pairs
+func (l *logrusLogger) With(args ...interface{}) hclog.Logger {
+	return &logrusLogger{
+		impl:  l.impl.WithFields(argsToFields(args...)),
+		level: l.level,
+	}
+}
+
+// Named creates a logger that will prepend the name string on the front of all messages.
+// If the logger already has a name, the new value will be appended to the current
+// name. That way, a major subsystem can use this to decorate all it's own logs
```

Comment:
its

Reply:
yep (copied from hclog, i would never make that mistake :)